app/org/sagebionetworks/bridge/models/healthdata/HealthDataUserConsent.java

```diff
@@ -0,0 +1,85 @@
+package org.sagebionetworks.bridge.models.healthdata;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.joda.time.DateTime;
+
+import org.sagebionetworks.bridge.json.DateTimeToStringSerializer;
+import org.sagebionetworks.bridge.json.JodaDateTimeDeserializer;
+
+/**
+ * User consent metadata stored with the health data record. We store this in the Health Data Record table, so we can
+ * avoid doing a bunch of extra calls to other tables during export.
+ */
+@JsonDeserialize(builder = HealthDataUserConsent.Builder.class)
+public class HealthDataUserConsent {
```

Comment:
Based on the email thread, I *think* the consensus we reached is that we do want to tag each health data record with sharing prefs at the time of record creation. If the user flips their sharing prefs, we don't retroactively change the sharing prefs on the record. (The two big reasons are (a) it's more user-friendly and more intuitive to *not* retroactively export data when the user flips from not sharing to sharing and (b) if for any reason we need to re-export, we need a historical record of when the user was and wasn't sharing.)

Re: code maintenance - I agree 100%. Given more time, I would have preferred to use the same class shared across both APIs. Unfortunately, this is a non-trivial refactor that I'm afraid we don't have time for right now.